src/main/java/teammates/ui/template/FeedbackResponseCommentRow.java

```diff
@@ -36,22 +41,31 @@
     private boolean isInstructorAllowedToDelete;
     private boolean isInstructorAllowedToEdit;
 
-    public FeedbackResponseCommentRow(FeedbackResponseCommentAttributes frc, String giverDisplay) {
+    public FeedbackResponseCommentRow(FeedbackResponseCommentAttributes frc, String giverDisplay,
+                                     Map<String, String> instructorEmailNameTable) {
+        this.instructorEmailNameTable = instructorEmailNameTable;
         this.commentId = frc.getId();
         this.giverDisplay = giverDisplay;
         this.createdAt = TimeHelper.formatDateTimeForComments(frc.createdAt);
-        this.editedAt = frc.getEditedAtText("Anonymous".equals(giverDisplay));
         this.commentText = frc.commentText.getValue();
+        setCommentGiverName(giverDisplay);
+        setCommentLastEditorName(frc.lastEditorEmail);
+        this.questionId = frc.feedbackQuestionId;
+        this.editedAt = setEditedAtText(frc.createdAt, frc.lastEditedAt);
     }
 
-    // for editing / deleting comments
     public FeedbackResponseCommentRow(FeedbackResponseCommentAttributes frc, String giverDisplay,
-                                      String giverName, String recipientName, String showCommentToString,
-                                      String showGiverNameToString,
-                                      Map<FeedbackParticipantType, Boolean> responseVisibilities) {
-        this(frc, giverDisplay);
+                                     String giverName, String recipientName, String showCommentToString,
+                                     String showGiverNameToString,
+                                     Map<FeedbackParticipantType, Boolean> responseVisibilities,
+                                     Map<String, String> instructorEmailNameTable) {
+        this(frc, giverDisplay, instructorEmailNameTable);
         setDataForAddEditDelete(frc, giverName, recipientName,
                                 showCommentToString, showGiverNameToString, responseVisibilities);
+        setCommentGiverName(giverDisplay);
+        setCommentLastEditorName(frc.lastEditorEmail);
+        this.questionId = frc.feedbackQuestionId;
+        this.editedAt = setEditedAtText(frc.createdAt, frc.lastEditedAt);
```

Comment:
There may not be a need for these 4 lines, considering they are already set after `this(frc, giverDisplay, instructorEmailNameTable);`